server/src/main/java/io/littlehorse/common/model/wfrun/ThreadRun.java

```diff
@@ -905,6 +908,49 @@ public void putVariable(String varName, VariableValue var) throws LHVarSubError
         }
     }
 
+    /**
+     * Creates a new variable on the current ThreadRun or any of its parents depending on
+     * who has the variable on its definition
+     * @param varName name of the variable
+     * @param var value of the variable
+     * @throws LHVarSubError when the varName is not found either on the current ThreadRun definition
+     * or its parents definition
+     */
+    public void createVariable(String varName, VariableValue var)
+        throws LHVarSubError {
+        VariableModification createVariable = (wfRunId, threadRunNumber, wfRun) -> {
+            Variable variable = new Variable(
+                varName,
+                var,
+                wfRunId,
+                threadRunNumber,
+                wfRun.getWfSpec()
+            );
+            wfRun.getDao().putVariable(variable);
+        };
+        applyOnAppropriateThread(varName, createVariable);
+    }
+
+    /**
+     * Mutates an existing variable on the current ThreadRun or any of its parents depending on
+     * who has the variable on its definition
+     * @param varName name of the variable
+     * @param var value of the variable
+     * @throws LHVarSubError when the varName is not found either on the current ThreadRun definition
+     * or its parents definition
+     */
+    public void mutateVariable(String varName, VariableValue var)
+        throws LHVarSubError {
+        VariableModification mutateVariable = (wfRunId, threadRunNumber, wfRun) -> {
+            Variable variable = wfRun
```

Comment:
Is it safe to assume that given we want to mutate the variable it will exists either on the `uncommittedChanges` or the `localStore`? I can make sure to throw an exception if the variable returned by `getDao().getVariable(...)` is null.
cc: @coltmcnealy-lh

Reply:
Thanks :)